security/subsystem/src/main/java/org/jboss/as/security/SecurityDomainResourceDefinition.java

```diff
@@ -226,19 +227,21 @@ protected void executeRuntimeStep(OperationContext context, ModelNode operation)
      * @throws OperationFailedException if the service is not available, or the thread was interrupted.
      */
     private static void waitForService(final ServiceController<?> controller) throws OperationFailedException {
-        if (controller.getState() == ServiceController.State.UP) return;
-
-        final StabilityMonitor monitor = new StabilityMonitor();
-        monitor.addController(controller);
         try {
-            monitor.awaitStability(100, MILLISECONDS);
+            final CountDownLatch latch = new CountDownLatch(1);
+            final LifecycleListener listener = new LifecycleListener() {
+                @Override
+                public void handleEvent(ServiceController<?> controller, LifecycleEvent event) {
+                    latch.countDown();
+                    controller.removeListener(this);
+                }
+            };
+            controller.addListener(listener);
+            latch.await();
```

Comment:
AIUI this method will now wait indefinitely. The method javadoc says 100ms, so if that is going to change the javadoc should.

The concern with waiting indefinitely is it could hang the management operation in the management thread and with the exclusive management op lock held. The OperationContext would never time out on its own, as the thread that does that kind of thing is the one blocking. Either the client would have to cancel the op (if it's using remoting) or the user would need to use the management API to find and terminate the hanging op.

It's probably ok. The scenario where this could happen is if this is called in the middle of a composite op where 

step 1 installs the security domain, but not something it depends on
step 2 invokes this
step 3 installs whatever the security domain depends on

An unlikely scenario.

Before that would fail here; now it will hang. Presumably anyone who was doing that saw the failures and fixed their logic.

Reply:
I returned it back to previous value.